test/e2e/set/scyllacluster/scyllacluster_pv.go

```diff
@@ -99,10 +116,7 @@ var _ = g.Describe("ScyllaCluster Orphaned PV controller", func() {
 		// ScyllaCluster is going to request a PVC from that StorageClass, and the test is going to request a clone of the original PVC
 		// from the default StorageClass to get any storage. Then the bound PV is rebounded to the original PVC
 		// but with empty NodeAffinity. This allows the test to trigger the orphaned PV cleanup logic.
-		provisionerCtx, provisionerCancel := context.WithCancel(ctx)
-		defer provisionerCancel()
 
-		var wg sync.WaitGroup
 		wg.Add(1)
```

Comment:
Why not swapping context creation and sync.WaitGroup? Context is created prematurely, and defer with Wait is in wrong place too.
Like this:
```
		var wg sync.WaitGroup
		defer wg.Wait()

		provisionerCtx, provisionerCancel := context.WithCancel(ctx)
		defer provisionerCancel()
		wg.Add(1)
		go func() {
			defer wg.Done()
```

Reply:
I don't mind - it seems a little cleaner. Changed.